src/walk.rs

```diff
@@ -48,11 +48,12 @@ pub enum FileType {
 /// If the `--exec` argument was supplied, this will create a thread pool for executing
 /// jobs in parallel from a given command line and the discovered paths. Otherwise, each
 /// path will simply be written to standard output.
-pub fn scan(root: &Path, pattern: Arc<Regex>, config: Arc<FdOptions>) {
+pub fn scan(path_vec: &mut Vec<PathBuf>, pattern: Arc<Regex>, config: Arc<FdOptions>) {
+    let first_path_buf = path_vec.pop().unwrap();
```

Comment:
Assuming that there is one element in the vector is fine for now. It would be even better to use the type system for that kind of guarantee (e.g. using a non-empty list type).

Reply:
Is there such a data structure native to rust? A non-empty vector is a precondition to scan so we could just include an assert statement to ensure that condition.

Reply:
I don't think so. There seem to be some nonempty crates, but if this is the only occurrence, I think your solution is fine. Instead of using an assert, we could replace `unwrap` with `expect` and a custom error message.